nuclia/sdk/auth.py

```diff
@@ -184,33 +191,29 @@ def post_login(self):
         self.accounts()
         self.zones()
 
-    def post_user(self, path: str, payload: Dict[str, str]):
-        if not self._config.token:
-            raise NeedUserToken()
-        resp = requests.post(
-            path,
-            headers={"Authorization": f"Bearer {self._config.token}"},
-            data=payload,
-        )
-        if resp.status_code == 201:
-            return resp.json()
-        elif resp.status_code == 403:
-            raise UserTokenExpired()
-
-    def get_user(self, path: str):
+    def _request(self, method: str, path: str, data: Optional[Any] = None):
         if not self._config.token:
             raise NeedUserToken()
-        resp = requests.get(
+        kwargs = {"headers": {"Authorization": f"Bearer {self._config.token}"}}
+        if data is not None:
+            non_null_values = {k: v for k, v in data.items() if v is not None}
+            kwargs["data"] = json.dumps(non_null_values)
+        resp = requests.request(
+            method,
             path,
-            headers={"Authorization": f"Bearer {self._config.token}"},
+            **kwargs,
         )
-        if resp.status_code == 200:
+        if resp.status_code == 204:
+            return None
+        elif resp.status_code >= 200 and resp.status_code < 300:
```

Comment:
I'm not sure there is a use-case right now, but 302 redirects would be raising an exception.